kernel/thread.c

```diff
@@ -145,6 +145,10 @@ int z_impl_k_thread_name_set(k_tid_t thread, const char *str)
 	strncpy(thread->name, str, CONFIG_THREAD_MAX_NAME_LEN - 1);
 	thread->name[CONFIG_THREAD_MAX_NAME_LEN - 1] = '\0';
 
+#ifdef CONFIG_ARCH_HAS_THREAD_NAME_HOOK
+	arch_thread_name_set(thread, str);
+#endif /* CONFIG_ARCH_HAS_THREAD_NAME_HOOK */
```

Comment:
I believe `IS_ENABLED` can be used here if you don't guard the declaration (which is a good reason not to guard declarations).